client/indexer-db/src/models/file.rs

```diff
@@ -0,0 +1,115 @@
+use chrono::NaiveDateTime;
+use diesel::prelude::*;
+use diesel_async::RunQueryDsl;
+
+use crate::{
+    models::PeerId,
+    schema::{file, file_peer_id},
+    DbConnection,
+};
+
+pub enum FileStorageRequestStep {
+    Requested = 0,
+    Fullfilled = 1,
+    Expired = 2,
+    Revoked = 3,
+}
+
+/// Table that holds the Files (both ongoing requests and completed).
+#[derive(Debug, Queryable, Insertable, Selectable)]
+#[diesel(table_name = file)]
+pub struct File {
+    /// The ID of the file as stored in the database. For the runtime id, use `onchain_bsp_id`.
+    pub id: i32,
+    pub account: String,
+    pub file_key: Vec<u8>,
+    pub bucket_id: i32,
+    pub location: Vec<u8>,
+    pub fingerprint: Vec<u8>,
+    pub size: i64,
+    /// The step this file is at. 0 = requested, 1 = fullfilled.
+    pub step: i32,
+    pub created_at: NaiveDateTime,
+    pub updated_at: NaiveDateTime,
+}
+
+/// Association table between File and PeerId
+#[derive(Debug, Queryable, Insertable, Associations)]
+#[diesel(table_name = file_peer_id)]
+#[diesel(belongs_to(File, foreign_key = file_id))]
+#[diesel(belongs_to(PeerId, foreign_key = peer_id))]
+pub struct FilePeerId {
+    pub file_id: i32,
+    pub peer_id: i32,
+}
+
+impl File {
+    pub async fn create<'a>(
+        conn: &mut DbConnection<'a>,
+        account: impl Into<String>,
+        file_key: impl Into<Vec<u8>>,
+        bucket_id: i32,
+        location: impl Into<Vec<u8>>,
+        fingerprint: impl Into<Vec<u8>>,
+        size: i64,
```

Comment:
Can we use `FileStorageRequestStep` as we do in `update_step`?

Reply:
Not sure what you are referring to.